app/src/test/java/io/appium/uiautomator2/model/settings/AbstractSettingTests.java

```diff
@@ -25,52 +25,90 @@
 
 import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
 
+import static org.mockito.ArgumentMatchers.anyInt;
 import static org.mockito.ArgumentMatchers.anyLong;
 import static org.mockito.Mockito.doThrow;
 import static org.mockito.Mockito.verify;
 import static org.mockito.MockitoAnnotations.initMocks;
 
 public class AbstractSettingTests {
-
     @Rule
     public ExpectedException thrown = ExpectedException.none();
 
     @Spy
-    private DummySetting dummySetting;
+    private DummyIntegerSetting dummyIntegerSetting;
+
+    @Spy
+    private DummyLongSetting dummyLongSetting;
 
     @Before
     public void setUp() {
         initMocks(this);
     }
 
+    @Test(expected = UiAutomator2Exception.class)
+    public void dummyIntegerSettingShouldThrowExceptionIfTypeIsNotValid() {
+        dummyIntegerSetting.update("test");
+    }
+
     @Test
-    public void shouldThrowExceptionIfTypeIsNotValid() {
-        thrown.expect(UiAutomator2Exception.class);
-        thrown.expectMessage("Invalid setting value type. Got: java.lang.String. Expected: java.lang.Long");
-        dummySetting.update("test");
+    public void dummyIntegerSettingShouldNotThrowExceptionIfApplyFailed() {
+        doThrow(new UiAutomator2Exception("error")).when(dummyIntegerSetting).apply(anyInt());
+        dummyIntegerSetting.update(10);
     }
 
     @Test
-    public void shouldNotThrowExceptionIfApplyFailed() {
-        doThrow(new UiAutomator2Exception("error")).when(dummySetting).apply(anyLong());
-        dummySetting.update(10);
+    public void dummyIntegerSettingShouldReturnValidValueType() {
+        Assert.assertEquals(Integer.class, dummyIntegerSetting.getValueType());
     }
 
     @Test
-    public void shouldReturnValidValueType() {
-        Assert.assertEquals(Long.class, dummySetting.getValueType());
+    public void dummyIntegerSettingShouldCallApplyWithValidValue() {
+        dummyIntegerSetting.update(123);
+        verify(dummyIntegerSetting).apply(123);
+    }
+
+    @Test(expected = UiAutomator2Exception.class)
+    public void dummyLongSettingShouldThrowExceptionIfTypeIsNotValid() {
+        dummyLongSetting.update("test");
+    }
+
+    private class DummyIntegerSetting extends AbstractSetting<Integer> {
+        public DummyIntegerSetting() {
+            super(Integer.class, "dummyInteger");
+        }
+
+        @Override
+        public Integer getValue() {
+            return 123;
+        }
+
+        @Override
+        protected void apply(Integer value) {
+            // do nothing
+        }
     }
 
     @Test
-    public void shouldCallApplyWithValidValue() {
-        dummySetting.update(123);
-        verify(dummySetting).apply(123L);
+    public void dummyLongSettingShouldNotThrowExceptionIfApplyFailed() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dpgraham/appium-uiautomator2-server/pullRequest?prid=5553523)